### PR TITLE
Add support for json response in restore watch commands

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

_Jira ticket:_ [CLOUDP-189430](https://jira.mongodb.org/browse/CLOUDP-189430)

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

A customer has reported they get "null" as a response from watch commands. Not sure If I'm missing some context but I believe we could return the same phrase in json format. Putting this PR up as a sample of what I'd change in all other watchers. Looking for early review, if approved I'll update with all watchers.

```shell
~/workplace/mongocli CLOUDP-189430 *20 ❯ ./bin/atlas backup restore watch -P cloud-dev $ID  --clusterName Cluster0                                                                                          1.20 10:54:42

Restore completed.
~/workplace/mongocli CLOUDP-189430 *20 ❯ ./bin/atlas backup restore watch -P cloud-dev $ID  --clusterName Cluster0 --output json                                                                            1.20 10:58:03
{
  "cancelled": false,
  "deliveryType": "automated",
  "desiredTimestamp": {},
  "expired": false,
  "failed": false,
  "finishedAt": "2023-07-19T09:50:59Z",
  "id": "64b7b0d0fcf3d660737ef263",
  "links": [
    {
      "href": "https://cloud-dev.mongodb.com/api/atlas/v2/groups/61dc598fd1f07c7b37ba425d/clusters/Cluster0/backup/restoreJobs/64b7b0d0fcf3d660737ef263",
      "rel": "self"
    },
    {
      "href": "https://cloud-dev.mongodb.com/api/atlas/v2/groups/61dc598fd1f07c7b37ba425d/clusters/Cluster0/backup/snapshots/64b7af43fcf3d660737ed5fe",
      "rel": "https://cloud.mongodb.com/snapshot"
    },
    {
      "href": "https://cloud-dev.mongodb.com/api/atlas/v2/groups/61dc598fd1f07c7b37ba425d/clusters/Cluster0",
      "rel": "https://cloud.mongodb.com/cluster"
    },
    {
      "href": "https://cloud-dev.mongodb.com/api/atlas/v2/groups/61dc598fd1f07c7b37ba425d",
      "rel": "https://cloud.mongodb.com/group"
    }
  ],
  "snapshotId": "64b7af43fcf3d660737ed5fe",
  "targetClusterName": "Cluster1",
  "targetGroupId": "64b54b47b9429a67128033fa",
  "timestamp": "2023-07-19T09:41:17Z"
}```

### CODE REVIEW

File: internal/cli/atlas/backup/restores/watch.go [hunk 1]
```diff
@@ -59,7 +60,7 @@ func (opts *WatchOpts) Run() error {
 		return err
 	}
 
-	return opts.Print(nil)
+	return opts.Print(watcherSuccess)
```
Comment (gssbzn) [inline internal/cli/atlas/backup/restores/watch.go:63]: if we were to fix this the proper fix would be to return the object being watched

**Comment by wtrocki** [comment]
+100 for going in this direction. We will improve clarity and enable users to gather more insights from the watch command itself.

File: internal/cli/atlas/backup/restores/watch.go (hunk 1 above)
Comment (blva) [reply internal/cli/atlas/backup/restores/watch.go]: updated, thank you!

**Review by gssbzn — APPROVED** [review]
LGTM thanks

**Review by wtrocki — APPROVED** [review]

**Review by tibulca — APPROVED** [review]

**Pull request merged**